src/main/java/webserver/RequestHandler.java

```diff
@@ -1,55 +1,63 @@
 package webserver;
 
+import java.io.BufferedReader;
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.net.Socket;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import utils.RequestParser;
+import webserver.entity.RequestEntity;
+import webserver.entity.ResponseEntity;
+
 public class RequestHandler implements Runnable {
     private static final Logger logger = LoggerFactory.getLogger(RequestHandler.class);
-
+    private final Controller controller;
+    
     private Socket connection;
-
-    public RequestHandler(Socket connectionSocket) {
+    
+    public RequestHandler(Socket connectionSocket, Controller controller) {
         this.connection = connectionSocket;
+        this.controller = controller;
     }
-
+    
+    @Override
     public void run() {
         logger.debug("New Client Connect! Connected IP : {}, Port : {}", connection.getInetAddress(),
-                connection.getPort());
-
+            connection.getPort());
+        
         try (InputStream in = connection.getInputStream(); OutputStream out = connection.getOutputStream()) {
-            // TODO 사용자 요청에 대한 처리는 이 곳에 구현하면 된다.
-            DataOutputStream dos = new DataOutputStream(out);
-            byte[] body = "Hello World".getBytes();
-            response200Header(dos, body.length);
-            responseBody(dos, body);
+            handle(in, out);
         } catch (IOException e) {
             logger.error(e.getMessage());
         }
     }
-
-    private void response200Header(DataOutputStream dos, int lengthOfBodyContent) {
-        try {
-            dos.writeBytes("HTTP/1.1 200 OK \r\n");
-            dos.writeBytes("Content-Type: text/html;charset=utf-8\r\n");
-            dos.writeBytes("Content-Length: " + lengthOfBodyContent + "\r\n");
-            dos.writeBytes("\r\n");
-        } catch (IOException e) {
-            logger.error(e.getMessage());
-        }
+    
+    private void handle(final InputStream in, final OutputStream out) throws IOException {
+        InputStreamReader ir = new InputStreamReader(in);
+        BufferedReader br = new BufferedReader(ir);
+        DataOutputStream dos = new DataOutputStream(out);
+        
+        ResponseEntity responseData = service(br);
+        dos.writeBytes(responseData.toResponseMessage());
+        dos.flush();
```

Comment:
`BufferedReader`는 단순히 `BufferedReader` 객체를 생성하기위한 용도이니 변수할당을 해줄 필요는 없을 것 같아요. 🙂

```suggestion
        BufferedReader br = new BufferedReader(new InputStreamReader(in));
        DataOutputStream dos = new DataOutputStream(out);
        
        ResponseEntity responseData = service(br);
        dos.writeBytes(responseData.toResponseMessage());
        dos.flush();
```

Reply:
해당 피드백 반영했습니다! 544d5ee